mmcv/cnn/bricks/scale.py

```diff
@@ -19,3 +19,41 @@ def __init__(self, scale: float = 1.0):
 
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         return x * self.scale
+
+
+class LayerScale(nn.Module):
+    """LayerScale layer.
+
+    Args:
+        dim (int): Dimension of input features.
+        inplace (bool): Whether performs operation in-place.
+            Default: `False`.
+        data_format (str): The input data format, could be 'channels_last'
+            or 'channels_first', representing (B, C, H, W) and
+            (B, N, C) format data respectively. Default: 'channels_last'.
+        scale (float): Initial value of scale factor. Default: 1.0
+    """
+
+    def __init__(self,
+                 dim: int,
+                 inplace: bool = False,
+                 data_format: str = 'channels_last',
+                 scale: float = 1e-5):
+        super().__init__()
+        assert data_format in ('channels_last', 'channels_first'), \
+            "'data_format' could only be channels_last or channels_first."
+        self.inplace = inplace
+        self.data_format = data_format
+        self.weight = nn.Parameter(torch.ones(dim) * scale)
+
+    def forward(self, x) -> torch.Tensor:
+        if self.data_format == 'channels_first':
+            shape = tuple((-1, *(1 for _ in range(x.dim() - 2))))
+        else:
+            shape = tuple((*(1 for _ in range(x.dim() - 3)), -1))
```

Comment:
```
B, C, L = 4, 16, 9
weight = torch.ones(C)
x = torch.rand(B, C, L)
data_format = 'channels_first'

if data_format == 'channels_first':
    shape = tuple((-1, *(1 for _ in range(x.dim()-1))))
else:
    shape = tuple((*(1 for _ in range(x.dim() - 3)), -1))

a =  x * weight.view(*shape)

>> RuntimeError: The size of tensor a (4) must match the size of tensor b (16) at non-singleton dimension 0
```

Reply:
Ahhh, I got it, I forget the batch dim. WIll it be more proper to set shape as 

```python
shape = tuple((-1, 1, *(1 for _ in range(x.dim()-1))))
```

This could make sure the dim of weight is equal to the dim of x, which could make sure the dot multiplying performed as out expected.

Reply:
```
B, C, L = 4, 16, 9
weight = torch.ones(C)
x = torch.rand(B, C, L)
data_format = 'channels_first'

if data_format == 'channels_first':
    shape = tuple((1, -1, *(1 for _ in range(x.dim()-1))))
else:
    shape = tuple((*(1 for _ in range(x.dim() - 1)), -1))

a =  x * weight.view(*shape)

>> RuntimeError: The size of tensor a (4) must match the size of tensor b (16) at non-singleton dimension 1
```

Reply:
```
B, C, L = 4, 16, 9
weight = torch.ones(C)
x = torch.rand(B, C, L)
data_format = 'channels_first'

if data_format == 'channels_first':
    shape = tuple((-1, *(1 for _ in range(x.dim()-2))))
else:
    shape = tuple((*(1 for _ in range(x.dim() - 1)), -1))

a =  x * weight.view(*shape)
a.shape

>> torch.Size([4, 16, 9])

B, C, T, L = 4, 16, 2, 9
weight = torch.ones(C)
x = torch.rand(B, C, T, L)
data_format = 'channels_first'

if data_format == 'channels_first':
    shape = tuple((-1, *(1 for _ in range(x.dim()-2))))
else:
    shape = tuple((*(1 for _ in range(x.dim() - 1)), -1))

a =  x * weight.view(*shape)
a.shape

>> torch.Size([4, 16, 2, 9])
```

Reply:
My bad, it should be 
```python
shape = tuple((1, -1, *(1 for _ in range(x.dim()-2))))
```

Reply:
Ah, I understand.
Thank you.